### DISCUSSION TITLE
Scara support

### DISCUSSION THREAD

**Discussion by simonschmeisser** [ros-industrial/staubli_val3_driver#49] [Q&A] [answered]
Has anyone experimented with using/extending this driver for 4-axis scara robots? We're looking at the TS2-100 to be specific but that shouldn't matter in general

**Comment by gavanderhoorn** [comment]
See https://github.com/ros-industrial/staubli_experimental/issues/36.

**Comment by simonschmeisser** [comment]
Thanks for the link! Good to know that my optimistic guess of this being achievable was not too optimistic. If you still have the code around somewhere that would be great but the overview in that comment is also already a good start. Let's see if the project materializes first.

**Comment by gavanderhoorn** [comment]
I'll see if I can find it.

It should be somewhere on a laptop I used, but *where* ..

**Comment by gavanderhoorn** [comment]
I do remember the scara having more trouble with the acceleration issues we still have with `staubli_val3_driver` though.

**Comment by gavanderhoorn** [comment, marked as answer]
I believe https://github.com/ros-industrial/staubli_val3_driver/compare/master...gavanderhoorn:wip_scara_support should do it. There were more changes, but I don't believe they were necessary to get things to work motion-wise.

No guarantees of course.

And heed the 'warning':

> ugly, hacky ..

**Comment by simonschmeisser** [comment]
Thanks!

**Comment by gavanderhoorn** [comment]
Have you been able to verify this works @simonschmeisser ?

**Comment by simonschmeisser** [comment]
No, not yet unfortunately. The project is still in the planning phase currently and I have not yet seen any hardware unfortunately.

**Comment by gavanderhoorn** [comment]
Friendly ping

**Reply by simonschmeisser** [reply]
unfortunately we did not get the order

**Reply by gavanderhoorn** [reply]
Ok. Then I'm going to mark my answer with the link to the diff with my fork as *the* answer.
